src/tracking.ts

```diff
@@ -213,14 +213,18 @@ export class SearchIOAnalytics {
       if (lastEvent) {
         queryId = lastEvent.queryId;
       }
+      if (!queryId) {
```

Comment:
This is doing the opposite of what we want, I'm fairly sure.

This will error out on ad-hoc tracking events and _not_ error out on our `FUNNEL_ENTRY_TYPES`.

We should add a test to ensure we've got the logic correct here.